packages/wordpress-kit/src/lib/tailwindcssPlugin/utilities/Colors.ts

```diff
@@ -0,0 +1,211 @@
+import { colorList, gradientList } from '../Constants';
+
+/**
+ * Class to generate all the color related utilities
+ * @class ColorUtilities
+ * @param {ThemeType} theme - The theme object from tailwindcss plugin
+ *
+ * @example
+ * ```
+ * const color = new ColorUtilities(theme);
+ *
+ * color.getBackgroundUtilities() // returns the background utilities
+ * color.getBorderColorUtilities() // returns the border color utilities
+ * ```
+ *
+ */
+export class ColorUtilities {
+  private theme: ThemeType;
+
+  constructor(theme: ThemeType) {
+    this.theme = theme;
+  }
+
+  /**
+   * It check if the color is in the theme colors list,
+   * or if it's a taliwind default color or just return a hex color.
+   *
+   * @param color - A color object
+   * @returns string - A string with the color value
+   */
+  private getColor = (color: Color) =>
+    this.theme(
+      `colors.${color.themeName}`,
+      this.theme(`colors.${color.tailwindDefault}`, color.hexDefault)
+    );
+
+  /**
+   * This find a color by the themeName in the colorList and return
+   * a Color object, throwing an error on compilation if the color is not found.
+   *
+   * @param {string} colorName
+   * @return Color
+   * @throws Error(`Color ${colorName} not found`)
+   */
+  private findColor = (colorName: string) => {
+    const color = colorList.find(color => color.themeName === colorName);
+    if (!color) {
+      throw new Error(`Color ${colorName} not found`);
+    }
+    return color;
+  };
+
+  /**
+   * @private
+   * This format a list of colors from the gradients return
+   * a string with the colors and position separated by a comma, to be used in the
+   * linear-gradient css function.
+   * @param colors - Array of color names
+   * @return {string} '#000000 50%, #ffffff 100%'
+   */
+  private formatGradientColors = (colors: GradientColors[]) => {
+    const formattedColors = colors.map(({ color, position }) => {
+      if (color.startsWith('#')) {
+        return `${color} ${position}`;
+      }
+
+      const themeColor = this.findColor(color);
+      return `${this.getColor(themeColor)} ${position}`;
+    });
+
+    return formattedColors.join(', ');
+  };
+
+  /**
+   *
+   * This return the color utilities.
+   * It take the colors from the colorList constant and
+   * build an object with the following structure:
+   *
+   * @example
+   * const colorUtilities = getColorUtilities();
+   * ```
+   * {
+   *  '.has-primary-color': {
+   *   color: '#0070f3 !important',
+   *  },
+   *  ...
+   * },
+   * ```
+   */
+  getColorUtilities = () =>
+    colorList.reduce(
+      (acc, color) => ({
+        ...acc,
+        [`.has-${color.name}-color`]: {
+          color: this.getColor(color),
+        },
+      }),
+      {}
+    );
+
+  /**
+   *
+   * This return the background utilities.
+   *
+   *
+   * It take the colors from the colorList constant to
+   * build an object with the background colors and a padding
```

Comment:
Similar to the previous one, the first line can be made into a `@return` tag

```suggestion
   * Taks the colors from the colorList constant to
   * build an object with the background colors and a padding.
   *
   * @returns {type?} background utilities
```